### PR TITLE
Removed deprecated parameters from AzureKeyVaultBackend

### PR DESCRIPTION
**Pull request by gopidesupavan**
Removed deprecated parameters from AzureKeyVaultBackend

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: newsfragments/41536.significant.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+**Breaking Change**
+
+The ``get_conn_uri`` method has been removed from the ``AzureKeyVaultBackend`` class.
```
Comment (eladkal) [inline newsfragments/41536.significant.rst:3]: What is the motivation to do this breaking change?

Also providers are not part of core cycle so if we want to so this we need entry in the provider change log not a newsfragemnt (we dont have yet newsfragment for peoviders)

Comment (gopidesupavan) [reply newsfragments/41536.significant.rst:3]: Ah i see your correct, will remove this update this PR :) thank you

Comment (gopidesupavan) [reply newsfragments/41536.significant.rst:3]: was trying to to remove airflow3 deprecation warnings in BaseSecretsBackend , thought of first removing in from the usage places. Please suggest should i ignore doing these changes in providers?

Comment (gopidesupavan) [reply newsfragments/41536.significant.rst:3]: @eladkal this provider change is not required at present. can i close this pull request ?

**Pull request closed without merging**